### PR TITLE
feat: verification script

### PR DESCRIPTION
**Pull request by matt-user**

### CODE REVIEW

File: signature-verification/verification-script/tests/harness.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+#[tokio::test]
+async fn valid_signature_returns_true_for_validating() {
+    let fuel_wallet = launch_provider_and_get_wallet().await.unwrap();
+
+    // Create solana wallet
+    let solana_keypair = Keypair::new();
+    let solana_address = solana_keypair.pubkey();
+
+    // Create the predicate by setting the signer and pass in the witness agrument
```
Comment (K1-R1) [inline signature-verification/verification-script/tests/harness.rs]:
```suggestion
    // Create the script by setting the signer and pass in the witness argument
```

File: signature-verification/verification-script/tests/harness.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        .main(witness_index)
+        .with_tx_policies(TxPolicies::default().with_witness_limit(144).with_script_gas_limit(10000000));
+
+    let mut tx = script_call_handler.build_tx().await.unwrap();
+
+    // Now that we have the tx the solana wallet must sign the ID
+    let consensus_parameters = fuel_wallet.provider().unwrap().consensus_parameters();
+    let tx_id = tx.id(consensus_parameters.chain_id);
+
+    let signature = solana_keypair.sign_message(&(*tx_id));
```
Comment (K1-R1) [inline signature-verification/verification-script/tests/harness.rs:36]:
```suggestion
    let signature = solana_keypair.sign_message(tx_id);
```
Does this not work?

File: signature-verification/verification-script/tests/harness.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+#[tokio::test]
+async fn invalid_signature_returns_false_for_failed_validation() {
+    let fuel_wallet = launch_provider_and_get_wallet().await.unwrap();
+
+    // Create solana wallet
+    let solana_keypair = Keypair::new();
+    let solana_address = solana_keypair.pubkey();
+
+    // Create the predicate by setting the signer and pass in the witness agrument
```
Comment (K1-R1) [inline signature-verification/verification-script/tests/harness.rs]:
```suggestion
    // Create the script by setting the signer and pass in the witness argument
```

File: signature-verification/verification-script/tests/harness.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        .main(witness_index)
+        .with_tx_policies(TxPolicies::default().with_witness_limit(144).with_script_gas_limit(10000000));
+
+    let mut tx = script_call_handler.build_tx().await.unwrap();
+
+    // Now that we have the tx the solana wallet must sign the ID
+    let consensus_parameters = fuel_wallet.provider().unwrap().consensus_parameters();
+    let tx_id = tx.id(consensus_parameters.chain_id);
+
+    let signature = solana_keypair.sign_message(&(*tx_id));
```
Comment (K1-R1) [inline signature-verification/verification-script/tests/harness.rs:86]:
```suggestion
    let signature = solana_keypair.sign_message(tx_id);
```

**Review by K1-R1 — APPROVED** [review]
Very clean with `ed_verify`. Couple of small comments but looking good

File: signature-verification/verification-script/tests/harness.rs (hunk 2 above)
Comment (matt-user) [reply signature-verification/verification-script/tests/harness.rs:36]: this won't compile

**Pull request merged**